docs/book/src/developer/providers/cluster-infrastructure.md

```diff
@@ -27,6 +27,10 @@ A cluster infrastructure provider must define an API type for "infrastructure cl
             meant to be suitable for programmatic interpretation
         2. `failureMessage` (string): indicates there is a fatal problem reconciling the provider's infrastructure;
             meant to be a more descriptive value than `failureReason`
+        3. `failureDomains` (`failureDomains`): the failure domains that machines should be placed in. `failureDomains`
+            is a map, defined as `map[string]FailureDomainSpec`. `FailureDomainSpec` is defined as:
```

Comment:
We should mention that a unique identifier needs to be generated for each FailureDomain and be used as the key in the map

Reply:
whoops, this whole PR slipped off my radar. Thanks, will update